crates/client-ext/src/lib.rs

```diff
@@ -184,11 +187,22 @@ impl TryFrom<FullBlock> for SealedBlockWithMetadata {
             },
         };
 
+        #[cfg(not(feature = "fault-proving"))]
+        let header = partial_header
+            .generate(
+                transactions.as_slice(),
+                messages.as_slice(),
+                full_block.header.event_inbox_root.into(),
+            )
+            .map_err(|e| anyhow::anyhow!(e))?;
+
+        #[cfg(feature = "fault-proving")]
         let header = partial_header
             .generate(
                 transactions.as_slice(),
                 messages.as_slice(),
                 full_block.header.event_inbox_root.into(),
+                &ChainId::default(), // TODO: How do we get the ChainId here?
```

Comment:
Need to resolve this before marking the PR as ready.

Reply:
Done in https://github.com/FuelLabs/network-watchtower/pull/53/commits/86fa0c4e2d83298b40f468a25a967dba1b00f5c7